core/taskengine/engine_test.go

```diff
@@ -453,5 +453,323 @@ func TestTriggerCompletedTaskReturnError(t *testing.T) {
 	if err == nil || resultTrigger != nil {
 		t.Errorf("expect trigger error but succeed")
 	}
+}
+
+func TestCreateSecret(t *testing.T) {
```

Comment:
Currently we don’t need to add the same tests in sdk, if unit tests are done here, but I don’t have good visibility what are covered in the unit tests. 

For this module, let’s rely on these units for testing, but simple tests are still needed in the sdk, so I can look up the usage of the secrets sdk code.